api/v1alpha1/promotion_types.go

```diff
@@ -64,13 +74,37 @@ const (
 	PromotionStepStatusSkipped PromotionStepStatus = "Skipped"
 )
 
-// IsTerminal returns true if the PromotionPhase is a terminal one.
-func (p *PromotionPhase) IsTerminal() bool {
-	switch *p {
-	case PromotionPhaseSucceeded, PromotionPhaseFailed, PromotionPhaseErrored, PromotionPhaseAborted:
-		return true
+// Compare compares the severity of the current PromotionStepStatus with
+// another. It returns -1 if the current status is less severe than the other, 0
+// if the two statuses are equally severe, or 1 if the current status is more
+// severe than the other.
+func (s PromotionStepStatus) Compare(rhs PromotionStepStatus) int {
```

Comment:
Nit: slightly more idiomatic.

```suggestion
func (s PromotionStepStatus) Compare(other PromotionStepStatus) int {
```